theano/gof/type.py

```diff
@@ -634,7 +634,7 @@ def c_code(self, node, name, inputs, outputs, sub):
         """ % dict(ctype=self.rtype.ctype, out=outputs[0], inp=inputs[0])
 
     def c_code_cache_version(self):
-        return (0,)
+        return (0, self.rtype.version)
```

Comment:
I would way that yes we want to.  But we also need to make it part of the key for cases it isn't None (like for cudnn).